### PR TITLE
DD-1216 part 2: generating series (part 1) as single compound field

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes DD-1216 part 2: generating series (part 1) as single compound field

# Description of changes

* transforms DDM description with descriptionType=seriesInformation into citation.series.seriesInformation
* provenance -> notesText

_Omitting DD- prefixes below to prevent confusing jira links._

remaining
* part two for series: see DD issue 1237, implement single value compound fields in the dataverse library and use it
* part 3 for DD-1216: access rights -> terms of  access

# How to test

* [x] Check [manually](https://dans-knaw.github.io/dd-ingest-flow/dev/) for dataverse-API surprises like DD issue 1233 
  ingested metadata of e1b53c4f-da5b-49fc-8732-a5aec7225b46 with `poetry run ingest-flow-start-migration`
  successfully examined the metadata of the draft on `https://dar.dans.knaw.nl/`
  

# Related PRs

(Add links)

*

# Notify

@DANS-KNAW/dataversedans

### CODE REVIEW

File: src/main/java/nl/knaw/dans/ingest/core/service/mapper/DepositToDvDatasetMetadataMapper.java [hunk 1]
```diff
@@ -261,6 +264,18 @@ void processMetadataBlock(boolean deduplicate, Map<String, MetadataBlock> fields
         fields.put(title, block);
     }
 
+    private MetadataField compoundBuild(CompoundFieldBuilder compoundFieldBuilder) {
+        // TODO rewrite in the dataverse library
+        CompoundField compoundField = compoundFieldBuilder.build();
+        if (compoundField.isMultiple())
+            return compoundField;
```
Comment (ericdevries) [inline src/main/java/nl/knaw/dans/ingest/core/service/mapper/DepositToDvDatasetMetadataMapper.java:271]: Not sure what the styleguide says, but please add braces

File: src/main/java/nl/knaw/dans/ingest/core/service/mapper/DepositToDvDatasetMetadataMapper.java [hunk 2]
```diff
@@ -261,6 +264,18 @@ void processMetadataBlock(boolean deduplicate, Map<String, MetadataBlock> fields
...
+        else return new MetadataField(compoundField.getTypeClass(),compoundField.getTypeName(),false) {
+            public Map<String, SingleValueField> getValue() {
+                return compoundField.getValue().get(0);
```
Comment (ericdevries) [inline src/main/java/nl/knaw/dans/ingest/core/service/mapper/DepositToDvDatasetMetadataMapper.java:274]: Perhaps a check to see if getValue() returns null would be useful, or is it never null?

**Review by ericdevries — COMMENTED** [review]
Looks good!

**Comment by janvanmansum** [comment]
Since I could only resolve the merge conflicts locally, I have copied this PR to #60 and merged that. Closing this one.

**Pull request closed without merging**